src/txn/txn.c

```diff
@@ -1114,10 +1127,13 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
      * the update is also from the same prepared transaction, restore the update from history store
      * or remove the key.
      */
-    if (F_ISSET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS) &&
+    prepare_on_disk = F_ISSET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS) &&
```

Comment:
All the conditions don't map to the variable name. Better to create some more variables and use them.